src/coap_mbedtls.c

```diff
@@ -1270,8 +1273,134 @@ mbedtls_debug_out(void *ctx COAP_UNUSED, int level,
   coap_log(log_level, "%s:%04d: %s", file, line, str);
 }
 
+#if !COAP_DISABLE_TCP
+/*
+ * return +ve data amount
+ *        0   no more
+ *        -ve  Mbed TLS error
+ */
+static int
+coap_sock_read(void *ctx, unsigned char *out, size_t outl) {
```

Comment:
Is there a reason why this function duplicates a lot of code from `coap_socket_read()` instead of calling that function (analogous to `coap_sock_write()`).

Reply:
I agree that there is common code.

However, the issue is setting up all the MBED_ERR_* returns correctly in the wrapper coap_socket_read() function which have to be correct for Mbed TLS to behave as expected.  The primary issue is coap_socket_read() returning -1 when recv() returns 0 (which also indicates EAGAIN / EINTR) which coap_sock_read() needs to send back MBEDTLS_ERR_SSL_CONN_EOF to the caller of coap_sock_read().  Not immediately clear how to do this without potentially breaking other things.

Reply:
Okay, then better leave it as it is.